File: 1630-count-odd-numbers-in-an-interval-range/count-odd-numbers-in-an-interval-range.java
class Solution {
    public int countOdds(int low, int high) {
        int length = high - low + 1;

        if (length % 2 == 1 && (high % 2 == 1 && low % 2 == 1)) {
            return (int) Math.ceil((double) length / 2);
        } else if (length % 2 == 1 && (high % 2 != 1 && low % 2 != 1)) {
            return length / 2;
        } else if (length % 2 == 0 && ((high % 2 == 1 && low % 2 == 0) || (high % 2 == 0 && low % 2 == 1))) {
            return (int) Math.ceil((double) length / 2);
        }

        return length / 2; 
    }
}

File: 1630-count-odd-numbers-in-an-interval-range/count-odd-numbers-in-an-interval-range.py
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        if (high - low + 1) % 2 == 1 and (high % 2 == 1 and low % 2 == 1):
            return math.ceil((high - low + 1) / 2)
        elif (high - low + 1) % 2 == 1 and (high % 2 != 1 and low % 2 != 1):
            return (high - low + 1) // 2
        elif (high - low + 1) % 2 == 0 and ((high % 2 == 1 and low % 2 == 0) or (high % 2 == 0 and low % 2 == 1)):
            return math.ceil((high - low + 1) / 2)
            
        